pkg/cloudprovider/gce/gce.go

```diff
@@ -500,6 +507,15 @@ func (gce *GCECloud) ExternalID(instance string) (string, error) {
 	return strconv.FormatUint(inst.Id, 10), nil
 }
 
+// InstanceID returns the cloud provider ID of the specified instance.
+func (gce *GCECloud) InstanceID(instance string) (string, error) {
+	inst, err := gce.getInstanceByName(instance)
+	if err != nil {
+		return "", err
+	}
+	return gce.projectID + "/" + gce.zone + "/" + strconv.FormatUint(inst.Id, 10), nil
```

Comment:
Actually, I meant that we should use the instance name here. The numerical ID can't be used to operate on the instance via the API AFAIK.